### PR TITLE
Add signing for fee estimation docs

### PR DESCRIPTION
**Pull request by cptartur**
Closes #

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    transaction = await account.sign_invoke_transaction(calls=call, max_fee=0)
+
+    # Re-sign a transaction for fee estimation
+    estimate_transaction = await account.sign_for_fee_estimate(transaction)
+
+    # Transaction uses a version that cannot be executed on StarkNet
+    assert estimate_transaction.version == 1 + 2**128
+    assert estimate_transaction.signature != transaction.signature
+
+    # Get a fee estimate
```
Comment (drknzz) [inline starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py:19]:
```suggestion
    # Get a fee estimation
```

File: starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import pytest
+
+
+@pytest.mark.asyncio
+async def test_signing_fee_estimate(gateway_account, map_contract):
+    account = gateway_account
+    # docs: start
+    # Create a transaction
+    call = map_contract.functions["put"].prepare(key=10, value=20, max_fee=1000)
```
Comment (war-in) [inline starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py:9]: Is it necessary to show `max_fee` usage here?

File: starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    assert estimate_transaction.version == 1 + 2**128
+    assert estimate_transaction.signature != transaction.signature
+
+    # Get a fee estimate
+    estimate = await account.client.estimate_fee(transaction)
+    assert estimate.overall_fee > 0
+
+    # Use a new fee in original transaction
+    transaction = await account.sign_invoke_transaction(
+        calls=call, max_fee=estimate.overall_fee * 2
```
Comment (war-in) [inline starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py:25]: I do not like `* 2`. Why was it added?

File: docs/guide/signing.rst [hunk 4]
```diff
@@ -22,3 +22,14 @@ You can also **verify a message**, which is done by a call to ``is_valid_signatu
 .. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_sign_offchain_message.py
     :language: python
     :dedent: 4
+
+
+Signing for fee estimation
+--------------------------
+
+:ref:`Account` allows signing transactions only for the purpose of fee estimation.
```
Comment (drknzz) [inline docs/guide/signing.rst:30]:
```suggestion
:ref:`Account` allows signing transactions for the purpose of fee estimation.
```
With `only` it sounds to me like it's the only functionality.

File: docs/guide/signing.rst [hunk 5]
```diff
@@ -22,3 +22,14 @@ You can also **verify a message**, which is done by a call to ``is_valid_signatu
...
+These transactions use a version that cannot be executed on StarkNet.
```
Comment (drknzz) [inline docs/guide/signing.rst:31]:
```suggestion
These transactions use a version that makes them non-executable on StarkNet.
```

File: starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py (hunk 2 above)
Comment (cptartur) [reply starknet_py/tests/e2e/docs/guide/test_sign_for_fee_estimate.py:9]: It isn't, removed

**Review by drknzz — APPROVED** [review]

**Pull request merged**